### PR TITLE
fix venue and room counting

### PR DESCRIPTION
**Pull request by denisdimitrov**
W.I.P

fix venue and room counting by migrating the structure to object with timestamps

### CODE REVIEW

**Comment by thecadams** [comment]
Hi @denisdimitrov 
Gave this a go and it seems to uncount you from attendance on the playa map while also removing your avatar, very easy to reproduce, you able to take a look?

Also if you push changes you'll find I've made some updates including fixing the typo `retainAttendance`. Best not to force push but rather to rebase whatever you do, or else to pull latest changes on the branch.

File: src/components/templates/Playa/Playa.tsx [hunk 1]
```diff
@@ -644,7 +669,9 @@ const Playa = () => {
                       {venue?.roomVisibility === RoomVisibility.nameCount && (
                         <div className="playa-venue-title">{v?.name}</div>
                       )}
-                      <div className="playa-venue-people">{usersInVenue}</div>
+                      <div className="playa-venue-people">
+                        {usersInVen.length}
```
Comment (thecadams) [inline src/components/templates/Playa/Playa.tsx:673]: usersInVenue?

File: src/pages/VenuePage/VenuePage.tsx [hunk 2]
```diff
@@ -64,8 +73,62 @@ const VenuePage = () => {
...
+      clearInterval(interval);
+      if (retainAttendance && user && profile) {
+        updateUserProfile(user.uid, {
+          lastSeenIn: { ...prevLocations, [venueName]: 0 },
+          lastSeenAt: new Date().getTime(),
+          room: null,
+        });
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (thecadams) [inline src/pages/VenuePage/VenuePage.tsx:129]: Guessing this is the cause of the location unloading? Would be great not to have this in the final PR.

**Review by thecadams — COMMENTED** [review]
Couple of things needing a second look, but mostly pretty good! Tons of solid work here, almost to the finish line!

File: src/pages/VenuePage/VenuePage.tsx (hunk 2 above)
Comment (denisdimitrov) [reply src/pages/VenuePage/VenuePage.tsx:129]: The problem here is that profile needs to be added in the dependency array and whenever this logic updates the location of the user, the profile is changed as well, and then this reruns and then it becomes an infinite loop

Comment (denisdimitrov) [reply src/pages/VenuePage/VenuePage.tsx:129]: If you have any suggestions or if statements that can avoid this feel free to post them here.

Comment (thecadams) [reply src/pages/VenuePage/VenuePage.tsx:129]: Understood. This is hiding a bug and usually will prevent renders that should happen..

Comment (thecadams) [reply src/pages/VenuePage/VenuePage.tsx:129]: What about changing updateUserProfile to call a function that only needs to receive the new attendance. Something where profile is no longer referenced inside the hook.

const { profile } = useUser()
const updateAttendance(user, loc, ts) { updateUserProfile(user.uid, {...profile, [loc]: ts}) }

Then you can remove the dependency on profile that’s inside the hook.

**Pull request merged**